### PR TITLE
jgrss/warp cleanup

### PR DESCRIPTION
**Pull request by jgrss**
## What is this PR changing?
* Simplifies the `mosaic()` method and ensures support for multi-band mosaics
* Fixes bug in `to_raster()`, where `nodata` values were not being written to file
* Adds `bigtiff` keyword argument to `save()` method
* Updates write tests

### CODE REVIEW

**Comment by jgrss** [comment]
@mmann1123 if you have time, could you test this branch out with some of your data?

**Comment by mmann1123** [comment]
Sure

**Review by mmann1123 — APPROVED** [review]
Looks good. Working on my end. I also added bigtiff to .save() please take a look. I added comments in the places you should look.

File: src/geowombat/core/io.py [hunk 1]
```diff
@@ -786,7 +792,12 @@ def save(
         compress=compress,
         tiled=True if max(blockxsize, blockysize) >= 16 else False,
         sharing=False,
+        bigtiff="IF_SAFER",
```
[the remark this answers is not in the document]
Comment (mmann1123) [reply src/geowombat/core/io.py:795]: There's an issue here trying it out on my files and I'm seeing empty rasters after mosaicing. Unfortunately I need to sign off and can't dig into why

**Review by mmann1123 — APPROVED** [review]
I think the issues i was having had to do with hitting memory limits, mosaicing huge 18 band images must have been bonking it. I've tried to recreate the issues on reasonably large datasets and haven't had the same issue. I think this is working well for almost all purposes.

**Comment by mmann1123** [comment]
resolves #309

File: src/geowombat/backends/xarray_.py [hunk 2]
```diff
@@ -435,114 +444,112 @@ def mosaic(
...
+            src_nodata=ref_kwargs['nodata'],
+            dst_nodata=tmp_nodata,
+            dtype='float64',
+        )
+        .gw.mask_nodata()
+        for wo in warped_objects
+    ]
+
+    # Apply the reduction
+    darray = functools.reduce(
```
Comment (jgrss) [inline src/geowombat/backends/xarray_.py:492]: @mmann1123 this is the main change for mosaicking.

File: tests/test_write.py [hunk 3]
```diff
@@ -12,157 +13,181 @@
...
+
                 with gw.open(out_path) as tmp_src:
-                    self.assertTrue(src.equals(tmp_src))
-                    self.assertTrue(hasattr(tmp_src, 'TEST_METADATA'))
-                    self.assertEqual(tmp_src.TEST_METADATA, 'TEST_VALUE')
+                    # Compare array values
+                    self.assertTrue(data.equals(tmp_src))
+                    # Compare attributes
+                    self.assertTrue(
+                        data.gw.nodataval == tmp_src.gw.nodataval == NODATA
```
Comment (jgrss) [inline tests/test_write.py:54]: @mmann1123 new test to check that the `nodata` value was properly saved and read.

File: src/geowombat/core/io.py [hunk 4]
```diff
@@ -950,126 +962,125 @@ def to_raster(
...
 
     else:
         compress = False
 
-    if 'nodata' not in kwargs:
-        if isinstance(data.gw.nodata, int) or isinstance(
-            data.gw.nodata, float
-        ):
-            kwargs['nodata'] = data.gw.nodata
+    if kwargs.get("nodata") is None:
```
Comment (jgrss) [inline src/geowombat/core/io.py:1002]: @mmann1123 this is the fix for ensuring the `nodata` value gets saved. Everything else in this file is auto-formatting.

**Comment by jgrss** [comment]
@mmann1123 I made `bigtiff` a keyword argument, with a default of `None`. You should be able to pass `data.gw.save(..., bigtiff='IF_SAFER')`. Let me know if that doesn't work.

**Comment by mmann1123** [comment]
> @mmann1123 I made `bigtiff` a keyword argument, with a default of `None`. You should be able to pass `data.gw.save(..., bigtiff='IF_SAFER')`. Let me know if that doesn't work.

That works.

File: src/geowombat/core/geoxarray.py [hunk 5]
```diff
@@ -739,6 +739,7 @@ def save(
         num_workers: T.Optional[int] = 1,
         log_progress: T.Optional[bool] = True,
         tqdm_kwargs: T.Optional[dict] = None,
+        bigtiff: T.Optional[str] = None,
```
[the remark this answers is not in the document]
Comment (mmann1123) [reply src/geowombat/core/geoxarray.py:742]: Yes

**Comment by jgrss** [comment]
> I think the issues i was having had to do with hitting memory limits, mosaicing huge 18 band images must have been bonking it. I've tried to recreate the issues on reasonably large datasets and haven't had the same issue. I think this is working well for almost all purposes.

Were you referring to this?

> There's an issue here trying it out on my files and I'm seeing empty rasters after mosaicing. Unfortunately I need to sign off and can't dig into why

**Comment by jgrss** [comment]
We can postpone this merge if you want to diagnose the large raster mosaicking. Is the file too large to upload and share?

**Comment by mmann1123** [comment]
I think we merge this, but we can open a large mosaic issue. 
I was able to resolve my issue using `da.maximum()` Maybe it is possible I was triggering compute or persist.

**Pull request merged**